internal/controller/finbackup_controller_test.go

```diff
@@ -363,14 +362,59 @@ var _ = Describe("FinBackup Controller Reconcile Test", Ordered, func() {
 			Expect(err).ShouldNot(HaveOccurred())
 
 			ExpectNoBackupJob(ctx, k8sClient, finbackup)
+		})
+	})
 
-			By("checking if the FinBackup is not marked as ready when the PVC is in a different Ceph cluster")
-			Consistently(func(g Gomega) {
-				var fb finv1.FinBackup
-				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(finbackup), &fb)
-				g.Expect(err).ShouldNot(HaveOccurred())
-				g.Expect(fb.IsReady()).Should(BeFalse(), "FinBackup should not be ready")
-			}, "5s", "1s").Should(Succeed())
+	// CSATEST-1629
+	// Description:
+	//	Prevent backups from being created for non-Ceph PVCs.
+	//
+	// Arrange:
+	//   - Create a StorageClass with a non-Ceph provisioner.
+	//   - Create a PVC using that non-Ceph StorageClass.
+	//
+	// Act:
+	//   - Create a FinBackup targeting the non-Ceph PVC.
+	//
+	// Assert:
+	//   - The reconciler does not return any errors.
+	//   - The reconciler does not create a backup job.
+	Context("Prevent backups from being created for non-Ceph PVCs", func() {
+		var pvc2 *corev1.PersistentVolumeClaim
+		var pv2 *corev1.PersistentVolume
+		var finbackup *finv1.FinBackup
+
+		BeforeEach(func(ctx SpecContext) {
+			By("creating another namespace for non-Ceph tests")
+			nonCephNamespace := &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: "test-namespace-non-ceph"}}
+			Expect(k8sClient.Create(ctx, nonCephNamespace)).Should(Succeed())
+
+			By("creating a non-Ceph StorageClass")
+			nonCephStorageClass := NewRBDStorageClass("non-ceph", nonCephNamespace.Name, rbdPoolName)
+			nonCephStorageClass.Provisioner = "non-ceph-provisioner.csi.com"
+			Expect(k8sClient.Create(ctx, nonCephStorageClass)).Should(Succeed())
```

Comment:
NewRBDStorageClass is to create RBD volumes. It's better not to use this function in L393.